### PR TITLE
Fix #2874 part 1 - XML formatter support for IMvcBuilder

### LINKED ISSUE #2874: Add IMvcBuilder overloads for localization and XML formatters
See here: https://github.com/aspnet/Mvc/issues/2872#issuecomment-125750316

We don't have an `IMvcBuilder` startup experience for these pieces.

### PR DESCRIPTION
**Pull request by rynowak**

### CODE REVIEW

File: src/Microsoft.AspNet.Mvc.Formatters.Xml/DependencyInjection/MvcXmlMvcBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+using Microsoft.AspNet.Mvc;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.Framework.DependencyInjection
+{
+    /// <summary>
+    /// Extension methods for adding XML formatters to MVC.
+    /// </summary>
+    public static class MvcJsonMvcBuilderExtensions
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Formatters.Xml/DependencyInjection/MvcXmlMvcBuilderExtensions.cs]: will fix name.

File: src/Microsoft.AspNet.Mvc.Formatters.Xml/MvcXmlDataContractSerializerMvcOptionsSetup.cs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNet.Mvc.Formatters.Xml;
+using Microsoft.AspNet.Mvc.ModelBinding.Metadata;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.AspNet.Mvc
+{
```
Comment (kirthik) [inline src/Microsoft.AspNet.Mvc.Formatters.Xml/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: docs for the class and `ConfigureMvc` method. Same for `MvcXmlSerializerMvcOptionsSetup` class too.

**Comment by kirthik** [comment]
Do you plan to add unit tests for `MvcXmlMvcBuilderExtensions` class in subsequent parts?

File: src/Microsoft.AspNet.Mvc.Formatters.Xml/DependencyInjection/MvcXmlMvcBuilderExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        /// <param name="builder">The <see cref="IMvcBuilder"/>.</param>
+        /// <returns>The <see cref="IMvcBuilder"/>.</returns>
+        public static IMvcBuilder AddXmlSerializerFormatters([NotNull] this IMvcBuilder builder)
+        {
+            AddAddXmlSerializerFormatterServices(builder.Services);
+            return builder;
+        }
+
+        // Internal for testing.
+        internal static void AddAddXmlDataContractSerializerFormatterServices(IServiceCollection services)
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Formatters.Xml/DependencyInjection/MvcXmlMvcBuilderExtensions.cs]: Is it `AddAdd...` on purpose? Sounds funny to me.

File: src/Microsoft.AspNet.Mvc.Formatters.Xml/MvcXmlDataContractSerializerMvcOptionsSetup.cs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    public class MvcXmlDataContractSerializerMvcOptionsSetup : ConfigureOptions<MvcOptions>
+    {
+        public MvcXmlDataContractSerializerMvcOptionsSetup()
+            : base(ConfigureMvc)
+        {
+            Order = DefaultOrder.DefaultFrameworkSortOrder + 10;
```
Comment (Eilon) [inline src/Microsoft.AspNet.Mvc.Formatters.Xml/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: 10's a nice number...

Comment (IDisposable) [inline src/Microsoft.AspNet.Mvc.Formatters.Xml/MvcXmlDataContractSerializerMvcOptionsSetup.cs]: I wonder what it's based on

**Comment by rynowak** [comment]
> Do you plan to add unit tests for MvcXmlMvcBuilderExtensions class in subsequent parts?

I wasn't planning on it. It's one line of code that registers a service.

**Comment by rynowak** [comment]
Updated

**Comment by Eilon** [comment]
:shipit:

**Pull request closed without merging**